Code/GraphMol/Kekulize.cpp

```diff
@@ -316,9 +309,22 @@ bool kekulizeWorker(RWMol &mol, const INT_VECT &allAtms,
             (nbrBond->getIsAromatic() ||
              mol.getAtomWithIdx(curr)->getAtomicNum() == 0 ||
              mol.getAtomWithIdx(nbrIdx)->getAtomicNum() == 0)) {
-          opts.push_back(nbrIdx);
+          // in order to try and avoid making wedged bonds double, we will add
+          // this neighbor at the back of the options after this loop if the bond
+          // is wedged. otherwise we append it to the options directly
+          if (nbrBond->getBondDir() == Bond::BondDir::BEGINWEDGE ||
+              nbrBond->getBondDir() == Bond::BondDir::BEGINDASH) {
+            wedgedOpts.push_back(nbrIdx);
+          } else {
+            opts.push_back(nbrIdx);
+          }
```

Comment:
in principle, the wedged neighbors could be also just pushed to the back and the regular one to the front of `opts` directly, similarly to how neighbors were originally added to the stack, and the kekulization tests would pass, but this would alter the general results for the kekulization algorithm, causing a fair number of unrelated tests to fail (most of the failing tests apparently just hard-code a dependency on the specific kekule structure and would need to be updated, but I think some algorithms also produce different results.. it might be worth investigating, but it might also be out of scope atm, I don't know).

Reply:
yeah, I was trying to avoid changing the results of the kekulization for cases where there are no wedged bonds.

If we're going to change keulization results, I'd rather look into larger algorithmic changes

Reply:
Sure, it makes sense.